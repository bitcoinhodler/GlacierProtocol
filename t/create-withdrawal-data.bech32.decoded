{
  "txid": "c367589b640bd8fced43af960523592f70192a8776507de29cc7ee5e6c2ff18f",
  "hash": "0d1e352d2ff64e34d835c1ec55a15dfe6f468877543b00642ee74f1f1d3ae312",
  "version": 2,
  "size": 405,
  "vsize": 189,
  "weight": 756,
  "locktime": 0,
  "vin": [
    {
      "txid": "bf777ec9fee0d176062be7b824a1ea6c029c7fd859051469dbe78305eeb7beca",
      "vout": 0,
      "scriptSig": {
        "asm": "002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8",
        "hex": "22002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8"
      },
      "txinwitness": [
        "",
        "3044022000dea458d7e0c93d13c74ce0bfb5f516cf6db6acb6a9134a94bd05a956ec6b0702206d73836fc0885505ade2a1419779d33aa76f58ac28285ca080d8001386a64aa101",
        "304402202b4742bdbdcadf550dfcd11887386a5e19f3819a6f838245a5d928e068defbad02207563f004a3c77369a9fba2d9b62803c4f7993804650474f202ea62b9d9ee85d501",
        "522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.99889303,
      "n": 0,
      "scriptPubKey": {
        "asm": "0 ce62ba41a5fc35dff5d97534004f984d6a89797d",
        "desc": "addr(bcrt1qee3t5sd9ls6alawew56qqnucf44gj7tajk3232)#rfwfrq4n",
        "hex": "0014ce62ba41a5fc35dff5d97534004f984d6a89797d",
        "address": "bcrt1qee3t5sd9ls6alawew56qqnucf44gj7tajk3232",
        "type": "witness_v0_keyhash"
      }
    }
  ]
}
