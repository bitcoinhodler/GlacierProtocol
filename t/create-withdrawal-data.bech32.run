#!/bin/bash

# Withdraw to a bech32 address

set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2N93du8YobdgsHyu3qgBvSyhGUT52utMNeA
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
bcrt1qee3t5sd9ls6alawew56qqnucf44gj7tajk3232
1
0200000007648644a295b6fc0c8e819249e4c5c0c2d43923e83bf6001df874583e71aa3aba010000006a473044022072424676d1fa48e470aaf946678a30a5fcfa30d58d1c95858988dc9ccf32b1b702206d3ce93c673d6e96abf7da16a0f2fd64b46d24ccc51bbd056da6e7006cfd9b4f01210399521d283c5e8488e2f3ad202b81e2d94b93394e2f7070e2108c6787e01dc12cfffffffff62f222ff10e1052657d1b892431479c3323b3172326ba05698869664c3d18cb010000006a473044022078fc91fa33f3c6dfc33636450911fd07066edbbc7f1e1637bebfe1e5ef22859702207e0bd6fa5e45e0a019e6385eb5f9fd5d1379948cad76cc6ebd9a7180e0bc22c6012103779631e9978658fca1a451126c5d135257f86c1b9be347238696642a24326e50ffffffffbcdcace096c35d98ecd7c523e920022fbe906018a09ba1bab0303accecc88c69010000006a4730440220286520aad536d992028b160e94139f8d84d42d2641d3e4bf4a019c6f1a27a24302201429715e80dfab7783327abec145421b21dbedfc6a74cd71191f6b6a810974aa0121024b63d59adc7d7e33daca9fb41ee9fe0043261c2e33b007e8c270af5a3b3afbdbffffffffda75ae66be39ece1da069efa240b558b68ebc624f130ab2e173d78525253fd17010000006a4730440220325ea18929bb0f1d04233863a0e73cb0a97fb43566b03d3e6d749db6640096c5022052f938843b31af0d4269087e89191a300c73c1ea6e70af3e9e7ac870c08b4b5d0121035c0aa8bd8dc73ea38dd76c5884d0a4308426cdd88913937912154fffc9955f56ffffffff398f1b40af0bdcca74cd1cbb230fff2ccdcd16a772a538d4dbc925df06ab980f010000006a473044022075ad3aa02c87656c6b711d064dc0e25fecd1af02c70bdbcf2eda9e84f45cdee6022042eefa2e3fd669380c30c9d52c735f48371397dae0c44856003a00ff7d8dd2cd0121031c40c2de64e34296bdec16c829caba13e031908553c4c7d4b59146ec000dc690ffffffff6355e60582b4575853d8ad7bf414b08891142b69a35ab7b062d360bc677d6596010000006a47304402206f2c3d16605a7d1569f1afa42b32d143691da8117be82bf6429bc7145e075b3e02207d59400cfc00c164c5a5fe473584b285fa0689eb1920df9570c27af8323256b30121036bde76cbe1716d8456058da0fabc0d682e991f5bacf538e2128245d29715160effffffff3da216be0eff1bc9c8220249eec30b60b3275ed1278fb01e9e19925b3b037267010000006a47304402201bcf6d84b2be8be320b1f359beed92f557d14575250ac15d19d2537c805fcc2802203a74e88ec95dc56042c8ad97210b5227b9c191c44ec905866afb4d5f8807576f012103165567fab4feead4f7fbf5ec5e99abeb3776290eeddd451ec7e44dc4b48bc32affffffff01f937f4050000000017a914ad50fb2489353008e75b0193011ff1f4b76f798c8700000000
2
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
10
y

y
INPUT
