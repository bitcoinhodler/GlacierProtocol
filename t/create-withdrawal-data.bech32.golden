Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.99891193 BTC

How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00002210 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00002210, you have 0.99888983 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to bcrt1qee3t5sd9ls6alawew56qqnucf44gj7tajk3232 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.99889303 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.99891193 BTC in unspent inputs from cold storage address 2N93du8YobdgsHyu3qgBvSyhGUT52utMNeA
0.99889303 BTC going to destination address bcrt1qee3t5sd9ls6alawew56qqnucf44gj7tajk3232
Fee amount: 0.00001890

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4


Confirm? (y/n): 
Calculating transaction...


Raw signed transaction (hex):
02000000000101e88a77c2db2a5fa939a108aaf894d0939e91ca40acd9393c60bb263b25359d07000000002322002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8ffffffff019730f40500000000160014ce62ba41a5fc35dff5d97534004f984d6a89797d040047304402201c042c5f241dc68b194a49eca235856f20234a9d8f610c2b3f44f8727fd6db43022001224d1ca09b53da82301fc11963c544fb349d2dd189a1eebd8a491e515be59201473044022058add16c858e8c2cde4ceed4962c39cadd5206fa6e438667b81786e3401bb43c022066af2ac2ae6289accff82971decd20529870d8cf8fad884dc637fef1100f8350018b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae00000000

Transaction fingerprint (md5):
0e5e4977dc56a2604b4dd6a0a07f90d8
QR code for transaction written to transaction.png
